crates/tx5-core/src/file_check.rs

```diff
@@ -17,78 +18,71 @@ impl FileCheck {
     }
 }
 
-fn get_user_cache_dir() -> Option<std::path::PathBuf> {
-    match app_dirs2::app_root(
+fn get_user_cache_dir() -> Result<std::path::PathBuf> {
+    app_dirs2::app_root(
         app_dirs2::AppDataType::UserCache,
         &app_dirs2::AppInfo {
             name: "host.holo.tx5",
             author: "host.holo.tx5",
         },
-    ) {
-        Ok(dir) => Some(dir),
-        _ => None,
-    }
+    )
+    .map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err))
```

Comment:
Should there be a custom tx5 error type that std::io::Error is passed up into?

It feels wrong to pass an option up and then do nothing if it is None. I think we should always be able to get a cache directory and not being able to do so should error.

Reply:
```suggestion
    .map_err(|err| std::io::Error::other(err))
```

Reply:
I've just been moving mostly towards using std::io::Error in all my rust code for interoperability. If an api needs to match on a specific variant, you can [downcast](https://doc.rust-lang.org/std/io/struct.Error.html#method.downcast) which admittedly is a little awkward, but not as bad in my opinion as libs all using a million different error types or anyhow or whatever.